### PR TITLE
Various cleanup and prepare for 1.0 release

### PR DESCRIPTION
**Pull request by takluyver**
Updating docs, package metadata and dependencies.

### CODE REVIEW

**Comment by takluyver** [comment]
The docs are still at pep517.readthedocs.io at the moment; should we set up a pyproject-hooks.pypa.io subdomain for them?

**Comment by takluyver** [comment]
Rebased; I'll likely merge this and release 1.0 after the weekend, if there are no objections. :slightly_smiling_face:

File: docs/changelog.rst [hunk 1]
```diff
@@ -1,6 +1,21 @@
 Changelog
 =========
 
+v1.0
+---
```
Comment (pradyunsg) [inline docs/changelog.rst:5]:
```suggestion
v1.0
----
```

Otherwise, it's not a heading.

Comment (takluyver) [reply docs/changelog.rst]: Thanks, good catch

**Pull request merged**